### ISSUE TITLE
Support for DayofWeek Function

### ISSUE THREAD

**Issue by OpusTen** [jpmml/sklearn2pmml#129]
Hi, 

I'm trying to get Day of the week from a date column in pandas.
This is actually part of a pipeline that I'm going to implement.
However, I'm facing the following issues. 

1. On saving the pipeline, I'm getting the following error.

RuntimeError: The JPMML-SkLearn conversion application has failed. The Java executable should have printed more information about the failure into its standard output and/or standard error streams

I suspect the reason to be lack of support for Pandas.dt.weekday_name function in PMML file format.
I'm aware that one needs to use 'DataFrameMapper' and use the functions supported by DataFrameMapper to be able to save the pipeline, but I couldn't find any example which does it.

**Can someone provide me a way to implement weekday_name functionality using a DataFrameMapper ?**
Thanks in advance.
Attaching Reproducible Code
[test_code.txt](https://github.com/jpmml/sklearn2pmml/files/2786226/test_code.txt)

.

**Comment by vruusmann** [comment]
The PMML language has decent temporal data type/functions support:

- http://dmg.org/pmml/v4-3/DataDictionary.html (see all those "date"-prefixed data types)
- http://dmg.org/pmml/v4-3/BuiltinFunctions.html#datedayssinceyear
- http://dmg.org/pmml/v4-3/BuiltinFunctions.html#datesecondssinceyear

The above functionality is supported starting from SkLearn2PMML version 0.41.0 (released mid-January 2019).

Here's your current "DayOfWeek" implementation:
``` python
base = datetime.datetime.today()
date_list = [base - datetime.timedelta(days=x) for x in range(0, numdays)]
```

The same, expressed using SkLearn2PMML transformers:
```python
from sklearn2pmml.decoration import Alias, DateDomain
from sklearn2pmml.preprocessing import DaysSinceYearTransformer, ExpressionTransformer

mapper = DataFrameMapper([
  ("date_column", [DateDomain(), DaysSinceYearTransformer(year = 2019), Alias(ExpressionTransformer("(X[0] % 7) - 2"), "int(day_of_week)", prefit = True)])
])
```

TLDR: The "DayOfWeek" is implemented by:

1. Calculating the number of days between the specified date and the 2019-01-01
2. Obtaining a day of week *relative to* 2019-01-01 by computing `$step1 % 7`
3. Adjusting the week offset day according to local traditions (eg. sunday in US, monday here in Estonia) by applying the necessary adjustment `$step2 - 2`

**Comment by vruusmann** [comment]
> I want to take the value "2019" from a dataframe column,

Cannot be done - it goes against the overall spirit/style of Scikit-Learn (ie. no other SkLearn transformer supports such "dynamic parameterization").

The reference year can be any year after 1900. Just make sure, that the `DaysSinceYear` field doesn't overflow accidentally.
